packages/core/src/controllers/relayer.ts

```diff
@@ -219,80 +252,74 @@ export class Relayer extends IRelayer {
     this.events.removeListener(event, listener);
   }
 
-  public async transportClose() {
-    // wait for all requests to finish before closing the transport
-    if (this.requestsInFlight.size > 0) {
-      this.logger.debug("Waiting for all in-flight requests to finish before closing transport...");
-      this.requestsInFlight.forEach(async (value) => {
-        await value.promise;
-      });
+  public async transportDisconnect() {
+    if (!this.hasExperiencedNetworkDisruption && this.connected && this.requestsInFlight.size > 0) {
+      try {
+        await Promise.all(
+          Array.from(this.requestsInFlight.values()).map((request) => request.promise),
+        );
+      } catch (e) {
+        this.logger.warn(e);
```

Comment:
Do we need an explicit error message here or do the promises return the wanted context if they throw?

Reply:
we don't expect errors but its just in case. Also ts complains on empty try/catch